client/allocrunner/alloc_runner_test.go

```diff
@@ -107,10 +107,19 @@ func TestAllocRunner_TaskLeader_KillTG(t *testing.T) {
 		}
 
 		found := false
+		killingMsg := ""
 		for _, e := range state1.Events {
 			if e.Type != structs.TaskLeaderDead {
 				found = true
 			}
+			if e.Type == structs.TaskKilling {
+				killingMsg = e.DisplayMessage
+			}
+		}
+		expectedKillingMsg := "Sent interrupt. Waiting 10ms before force killing"
+
+		if killingMsg != expectedKillingMsg {
```

Comment:
I'd suggest moving this check after `!found`.  Logically, we should only perform check if the kill event is found; if not found, we'd like a not found error message.